[REF] 8.0 mgmtsystem nonconformity

llacroix opened the pull request:
Removed base_status dependency and reimplemented some of the methods that were missing to keep the `cases` working. 

depends on
- [x] #22 
- [x] #35
  - [x] #26
    - [x] OCA/knowledge#30

llacroix committed:
Moved mgmtsystem_nonconformity to root for port

llacroix committed:
Updated module as installable and removed depdencie on base_status

See

the base_status module is apparently gone so we have to work without it.
https://code.launchpad.net/~openerp-dev/openobject-addons/trunk-remove-base_status-tde/+merge/179887

llacroix committed:
Small fix to use stage_id.name instead of state

State doesn't exist anymore

llacroix committed:
Use stage_id for actions

llacroix committed:
Add support for settled, rejected instead of done/canceled... quite ugly

bwrsandman commented on mgmtsystem_nonconformity/mgmtsystem_nonconformity.py:
```diff
@@ -22,7 +22,9 @@
 from openerp.osv import fields, orm
-from openerp.addons.base_status.base_state import base_state
+
+#from openerp.addons.base_status.base_state import base_state
```
Please remove this commented line

bwrsandman commented on mgmtsystem_nonconformity/mgmtsystem_nonconformity.py:
```diff
@@ -411,10 +413,11 @@ def wkf_open(self, cr, uid, ids, context=None):
             o.immediate_action_id.case_open()
         for a in o.action_ids:
-            if a.state == 'draft':
+            if a.stage_id.name.lower() == 'draft':
```
This will cause an error in cases where stage_id is not supplied

llacroix replied:
are you sure the stage_id can be null?

bwrsandman replied:
That doesn't seem to be checked.

bwrsandman commented on mgmtsystem_nonconformity/mgmtsystem_nonconformity.py:
```diff
@@ -22,7 +22,9 @@
 from openerp.tools.translate import _
```
Please split this file so it has one python file by model/class

bwrsandman commented:
As a general rule, unless the classes are very small (case of one2many). We strongly recommend you split python/xml files by model involved.

https://lists.launchpad.net/openerp-community/msg06955.html

llacroix committed:
Separated python in each model file

llacroix committed:
Added missing methodcase_close_send_note for workflows

llacroix committed:
Updated model severity to V8

llacroix committed:
Port cause model to v8

llacroix committed:
Fix flake8 error

llacroix committed:
Fix small error in `cause` and upgraded `origin` to v8

llacroix committed:
Splitted xml views into 5 different xml files

llacroix committed:
Moved icon.png to static/description

llacroix committed:
Added tests and updated cause to v8

llacroix committed:
Added test origin and changed name of test class for cause

llacroix committed:
Removed severity redefined here

llacroix committed:
Added tests for create nonconformity

llacroix committed:
Ported fields to v8

llacroix commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_cause.py:
```diff
@@ -0,0 +1,71 @@
+        res = self.name_get()
+        return dict(res)
+
+    _constraints = [
```
_constraint not present in mgmtsystem_nonconformity_origin.py. yet the code is almost the same. We could technically subclass both class and clone the implementation or just add a type instead of copying code.

bwrsandman replied:
Good catch on the constraint.

I haven't done this, but I think to clone a class you can do this:

``` python
class mgmtsystem_nonconformity_origin(models.Model):
    _name = "mgmtsystem.nonconformity.origin"
    _inherit = "mgmtsystem.nonconformity.cause"
```

If that doesn't work, you can use an abstract model and then subclass from that

llacroix replied:
Yes, but I'd rather subclass them as origin aren't causes and causes aren't origin. If it's fine to inherit only for the structure only then it could be fine but if there is a way to get all origins and it will catch all causes too it won't make sense.

bwrsandman commented:
depends on ~~#35~~

bwrsandman commented:
pep8:

```
./mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:79:8: E121 continuation line under-indented for hanging indent
```

llacroix committed:
Fix indent block

dreispt commented:
Could we have this to not depend on `document_page_procedure`?
We may want to manage NCs is Odoo but prefer to manage Procudure Docs elsewhere.

max3903 commented:
@dreispt having a dependency on document_page_procedure does not necessarily mean you manage your documentation content in Odoo. We use it as a list to reference the content in another application. The day we want to put everything in Odoo, we just have to transfer the content and drop the other application.

If you want to drop the dependency, you will have to create a new field to provide the list of procedures (quality), environmental aspects (environment), and other thing in other systems. 

I am sure you will agree with me that any Management Systems software should come with a documentation system, so why Odoo should be different here.

dreispt commented on mgmtsystem_nonconformity/board_mgmtsystem_nonconformity.xml:
```diff
@@ -3,6 +3,7 @@
     <data>
 
         <record id="action_mgmtsystem_nonconformity" model="ir.actions.act_window">
+            <field name="name">board.mgmtsystem.nonconformity.act_window</field>
```
name should be a human readable text

bwrsandman replied:
Or it can be ommited

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -151,8 +42,12 @@ class mgmtsystem_nonconformity_severity(orm.Model):
 
+own_company = lambda self: self.env.user.company_id.id
+default_date = lambda *a: time.strftime(DATE_FORMAT)
+default_user_id = lambda self: self.env.user.id
```
I bet assigning lambdas doesn't pass lint checks...

llacroix replied:
It shouldn't but the travis tests did pass. There might be something wrong with the tests.

llacroix replied:
Removed

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -162,127 +57,129 @@ class mgmtsystem_nonconformity(base_state, orm.Model):
-
-    def create(self, cr, uid, vals, context=None):
+    # 1. Description
+    id = fields.Integer('ID', readonly=True)
```
I think adding `id` is not needed anymore with the new API. Can you confirm?

llacroix replied:
Yes, as far as I can say, the id should be created by default even on openerp 7 as it is used in ir_model_data. Not sure how it worked before but it shouldn't be necessary. I'll test it thought.

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -162,127 +57,129 @@ class mgmtsystem_nonconformity(base_state, orm.Model):
+        readonly=True,
+        default="NEW"
+    )
+    date = fields.Date('Date', required=True, default=default_date)
```
I would rather have lambdas inline

llacroix replied:
Done, yet having named function makes it reusable. if we had a common module. The `own_company` is used in other modules but each copy the same method (lambda)

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -162,127 +57,129 @@ class mgmtsystem_nonconformity(base_state, orm.Model):
+    )
+    date = fields.Date('Date', required=True, default=default_date)
+    partner_id = fields.Many2one('res.partner', 'Partner', required=True)
+    reference = fields.Char('Related to', size=50)
```
drop size attribute?

llacroix replied:
Done

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -162,127 +57,129 @@ class mgmtsystem_nonconformity(base_state, orm.Model):
-            cr, uid, vals, context)
+        return super(mgmtsystem_nonconformity, self).create(vals)
 
     def message_auto_subscribe(
```
The messaging features need a rewrite. I'm not bothered by the direct port, but wouldn't mind if they were removed now.

llacroix replied:
You'll have to be more precise if you want something to get removed.

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -447,14 +346,17 @@ def wkf_cancel(self, cr, uid, ids, context=None):
         o = self.browse(cr, uid, ids, context=context)[0]
-        done_states = ['done', 'cancelled']
+        # TODO make it more friendly
+        done_states = ['done', 'cancelled', 'settled', 'rejected']
```
settled and rejected are not in the states list defined at the top

llacroix replied:
This is a problem, I wasn't sure how to handle this. The problem come from the fact that those states are on a `mgmtsystem.action` object. The `mgmtsystem.action` object is a `crm.claim object`. But the `crm.claim` object doesn't have states anymore but stages.

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -447,14 +346,17 @@ def wkf_cancel(self, cr, uid, ids, context=None):
+        done_states = ['done', 'cancelled', 'settled', 'rejected']
         if (o.immediate_action_id
-                and o.immediate_action_id.state not in done_states):
+                and o.immediate_action_id.stage_id.name.lower()
```
This is mixing up sates with states!
Something is not right here.

llacroix replied:
The object mgmtsystem.action doesn't have states anymore. I can add them back and ignore stages.

dreispt replied:
No, please keep Stages.

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_cause.py:
```diff
@@ -0,0 +1,71 @@
+    _description = "Cause of the nonconformity of the management system"
+    _order = 'parent_id, sequence'
+
+    id = fields.Integer('ID', readonly=True)
```
confirm if explicit id can be removed

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_cause.py:
```diff
@@ -0,0 +1,71 @@
+
+    @api.multi
+    def name_get(self):
+        reads = self.read(['name', 'parent_id'])
```
Don't use read. Iterate over the `self` recordset directly.

llacroix replied:
Done

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_cause.py:
```diff
@@ -0,0 +1,71 @@
+    """
+    _name = "mgmtsystem.nonconformity.cause"
+    _description = "Cause of the nonconformity of the management system"
+    _order = 'parent_id, sequence'
```
We should active hierarchical relations setting `_parent_store=True`. See https://www.odoo.com/documentation/8.0/reference/orm.html#openerp.models.Model._parent_store

llacroix replied:
we only have to add `_parent_store = True` ?

dreispt replied:
No, two helper fields also need to be added:

```
_parent_store = True
parent_left = fields.Integer('Parent Left', index=True)
parent_right = fields.Integer('Parent Right', index=True)
```

llacroix replied:
It finds by itself how to construct the left and right? I understand that it creates a structure that will allow fetching part of the trees efficiently but the doc doesn't tell much how it finds the left and right. I'll add it and lets see how it goes.

llacroix replied:
Added the fields.

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -21,8 +21,9 @@
 from openerp import netsvc
-from openerp.osv import fields, orm
-from openerp.addons.base_status.base_state import base_state
+from openerp.osv import orm
```
You shouldn't need this now...

llacroix replied:
Changed form `from openerp.exceptions import except_orm`

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_origin.py:
```diff
@@ -0,0 +1,65 @@
+from openerp import fields, models, api
+
+
+class mgmtsystem_nonconformity_origin(models.Model):
```
Same comments as for the `cause` model

llacroix replied:
Not sure to which comment you're referencing to.

llacroix replied:
@dreispt it's about the parent_id right?

llacroix replied:
Added the fields.

dreispt commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity_origin.py:
```diff
@@ -0,0 +1,65 @@
+    @api.multi
+    def _name_get_fnc(self, prop, unknow_none):
+        res = self.name_get()
+        return dict(res)
```
I don't think this does anything useful.

llacroix replied:
Removed that part

dreispt commented on mgmtsystem_nonconformity/tests/test_origin.py:
```diff
@@ -0,0 +1,35 @@
+
+        self.assertEqual(name_dict[record.id], "TestOrigin")
+        self.assertEqual(name_assoc[0][1], "TestOrigin")
+        self.assertEqual(name_assoc[0][0], record.id)
```
Kudos for the tests :trophy:

dreispt commented on mgmtsystem_nonconformity/views/mgmtsystem_cause.xml:
```diff
@@ -0,0 +1,62 @@
+            <field name="model">mgmtsystem.nonconformity.cause</field>
+            <field name="arch" type="xml">
+                <form string="Causes of nonconformity">
+                <field name="name"/>
```
I think you need a `<group>` element around the fields for the labels to be displayed.

llacroix replied:
Good point

dreispt commented on mgmtsystem_nonconformity/views/mgmtsystem_nonconformity.xml:
```diff
@@ -80,7 +80,7 @@
                             states="open" groups="mgmtsystem.group_mgmtsystem_manager,mgmtsystem.group_mgmtsystem_auditor"/>
                         <button name="case_reset" string="Reset to New" states="done,cancel" type="object"/>
-                        <field name="state" widget="statusbar" statusbar_visible="draft,analysis,pending,open,done"/>
+                        <field name="state" widget="statusbar" statusbar_visible="draft,analysis,pending,open,done" clickable="True"/>
```
I don't think you want this to be "clickable": flow needs to be controlled by logic in buttons

dreispt commented on mgmtsystem_nonconformity/views/mgmtsystem_origin.xml:
```diff
@@ -0,0 +1,63 @@
+            <field name="model">mgmtsystem.nonconformity.origin</field>
+            <field name="arch" type="xml">
+                <form string="Origins of nonconformity">
+                    <field name="name"/>
```
add `<group>` element

dreispt commented on mgmtsystem_nonconformity/views/mgmtsystem_severity.xml:
```diff
@@ -0,0 +1,32 @@
+            <field name="model">mgmtsystem.nonconformity.severity</field>
+            <field name="arch" type="xml">
+                <form string="Severity of Complaints and Nonconformities">
+                        <field name="name"/>
```
add `<group>` element

max3903 commented:
Needs rebase, I think.

llacroix commented:
Working on it now

dreispt commented:
@llacroix You are still to address my comments, right?

llacroix commented:
@dreispt yes I'm on it... I'll push updates

llacroix committed:
Removed ID and removed named lambdas

llacroix committed:
Simplify the name_get method

llacroix committed:
Get exceptions without importing orm

llacroix committed:
Removed size

llacroix committed:
Simplified name_get and removed id

llacroix committed:
Replace try except by self.assertRaise

llacroix committed:
Added group in forms

llacroix committed:
Reformat xml files removed tabs/space mix and 2 space indent

llacroix committed:
Fix tests

llacroix committed:
Added left and right to cause and origin

llacroix committed:
Add ondelete to parent_id

llacroix committed:
Fix tests

llacroix commented:
After looking at the current modules, I believe we might have to change `mgmtsystem.action` to use states instead. It might not be necessary to make it inherit from `crm.claim`

llacroix committed:
Improved name_get method and full coverage or origin/cause

bwrsandman commented on mgmtsystem_nonconformity/tests/test_nonconformity.py:
```diff
@@ -0,0 +1,44 @@
+from openerp.tests import common
```
Could you test the workflow too?

llacroix replied:
will do.

llacroix committed:
Added workflow test

llacroix committed:
Added missing copyrights

llacroix committed:
Fix flake errors

llacroix committed:
Fix more flake errors

llacroix committed:
Some actions can only be done once actions, analysis

llacroix committed:
Fix even more flakes errors

llacroix committed:
Test reset button

llacroix committed:
Added missing fields (used in demo data)

llacroix committed:
Test state nonconformity name

llacroix committed:
Remove states as we are using stages on actions...

It would make more sense to add a property to stages that mark some
stages as ending stages instead of checking by names.

llacroix committed:
Added stages to mgmtststem_action

llacroix committed:
Use stage is ending and is starting instead of by name

llacroix committed:
Test analysis instead of evaluation

llacroix committed:
Give a state in transition when case_open

llacroix committed:
Fix mgmtsystem_action tests

llacroix committed:
Fix flake8 errors

llacroix committed:
Flake error

llacroix committed:
Survey code doesn't exist anymore

llacroix committed:
Added base security

llacroix commented:
https://coveralls.io/files/678256153

bwrsandman commented:
@llacroix https://coveralls.io/builds/2466939/source?filename=mgmtsystem_nonconformity%2Fmodels%2Fmgmtsystem_nonconformity.py

llacroix committed:
Security rule in wrong project

bwrsandman commented on mgmtsystem_nonconformity/tests/test_workflows.py:
```diff
@@ -0,0 +1,222 @@
+
+        # Test Pending to Open
+        # pending -> analysis
+        # self.try_invalid_signal(conf_obj, 'button_analysis_p', 'analysis')
```
Dead code?

bwrsandman commented on mgmtsystem_nonconformity/tests/test_workflows.py:
```diff
@@ -0,0 +1,222 @@
+        # draft-> analysis (already on analysis)
+        self.try_signal(conf_obj, 'button_analysis_n', 'analysis')
+        # analysis -> pending
+        # self.try_invalid_signal(conf_obj, 'button_review_n', 'pending')
```
Dead code?

bwrsandman commented on mgmtsystem_nonconformity/tests/test_workflows.py:
```diff
@@ -0,0 +1,222 @@
+        # open -> done
+        self.try_invalid_signal(conf_obj, 'button_close', 'done')
+        # draft-> analysis
+        # self.try_invalid_signal(conf_obj, 'button_analysis_n', 'analysis')
```
Dead code?

bwrsandman commented on mgmtsystem_nonconformity/tests/test_workflows.py:
```diff
@@ -0,0 +1,222 @@
+                if rollback:
+                    raise BreakSavePoint()
+        except BreakSavePoint:
+            obj.refresh()
```
Why is it necessary to have a custom error? Is it because of the context manager? Maybe you could define the Exception within this function in order to keep it encapsulated.

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+
+    name = fields.Char('Name')
+
+    # 1. Description
```
This could be put in the helptext

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+    """Management System - Nonconformity."""
+
+    _name = "mgmtsystem.nonconformity"
+    _description = "Nonconformity of the management system"
```
Nonconformity of the Management System

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+
+class mgmtsystem_nonconformity(models.Model):
+
+    """Management System - Nonconformity."""
```
Docstring is redundant, if it doesn't explain what it is, it can be removed

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+                o = self.browse(cr, uid, ids, context=context)[0]
+                post = _(u'''
+<br />
+<ul><li> <b>State:</b> %s → %s</li></ul>\
```
Couldn't this be done simply by defining the track on the state column definition?

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+            self.message_post(cr, uid, [id], body=msg, context=context)
+        return True
+
+    def case_get_note_msg_prefix(self, cr, uid, id, context=None):
```
`@api.model`

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+##############################################################################
+from openerp.tools.translate import _
+from openerp import netsvc
+from openerp.exceptions import except_orm
```
These `except_orm` could be replaced by more specific exceptions
`excpetions` can be added to the `from openerp` line

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+#
+##############################################################################
+from openerp.tools.translate import _
+from openerp import netsvc
```
Could be added to line 24

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_nonconformity.py:
```diff
@@ -0,0 +1,425 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.tools.translate import _
```
You can import `_` directly `from openerp` on line 24

bwrsandman commented on mgmtsystem_nonconformity/models/mgmtsystem_action.py:
```diff
@@ -0,0 +1,42 @@
+        readonly=True,
+    )
+
+    # TODO add case_open/case_close conditions
```
Can an issue be added for this?

bwrsandman commented on mgmtsystem_action/tests/test_create_action.py:
```diff
@@ -11,7 +11,8 @@ def test_create_action(self):
         self.assertNotEqual(record.reference, "NEW")
         self.assertEqual(record.type_action, "immediate")
-        self.assertEqual(record.stage_id.name, "New")
+        self.assertEqual(record.stage_id.name, "Draft")
```
The use of `_()` would be more correct if it works.

dreispt commented:
Current status is needs fixing, correct?

bwrsandman commented:
@dreispt Yes, but unfortunately, @llacroix won't be working on it anymore.

Commit:
Move datas to their own folders

Commit:
Cleanup Nonconformity

Commit:
Use track and mail message subtypes instead of overriding functions

Commit:
Make admin a memeber of Approving User group

Commit:
Remove unused functions

Commit:
Cleanup and full use of v8 and track features

Commit:
Use a python context which does rollback

Commit:
Fix pep8

Commit:
Full test coverage of module

bwrsandman commented:
@max3903, @dreispt please test

dreispt commented:
Thanks @bwrsandman !

dreispt commented on mgmtsystem_action/mgmtsystem_action.py:
```diff
@@ -45,6 +45,18 @@ class mgmtsystem_action(models.Model):
+    stage_id = fields.Many2one(
+        'mgmtsystem.action.stage',
+        'Stage',
+        default=lambda self: self.get_default_stage()
```
You can use: `default=get_default_stage`

bwrsandman replied:
I use lambda so `get_default_stage` can be overriden. AFAIK setting default to a member function directly does not allow it to be overriden the way that a call to `self.` does.

dreispt commented on mgmtsystem_action/mgmtsystem_action.py:
```diff
@@ -45,6 +45,18 @@ class mgmtsystem_action(models.Model):
+    def get_default_stage(self):
+        return self.env['mgmtsystem.action.stage'].search([
+            ('is_starting', '=', True)
+        ]).id
```
In Projects, the logic used is to pick the first non-folded Stage. Maybe we could also fall-back to that is no initial stage is set?

bwrsandman replied:
Sure. Can you point me to the case in Projects?

dreispt replied:
I could some changes myself: 
I would like to add Stages, and with that remove a lot of code that implemented a hardcoded process flow.
Also, do you actually use wkf workflows? If you let me I'm tempted to drop them also.

dreispt replied:
I could some changes myself: 
I would like to add Stages, and with that remove a lot of code that implemented a hardcoded process flow.
Also, do you actually use wkf workflows? If you let me I'm tempted to drop them also.

bwrsandman replied:
What is gained from dropping them? Workflows can be altered by the user which I find very useful.

dreispt replied:
Changes in Workflow wiring are not reflected in the UI.
IMO Workflows are only useful for cross-Model processes (such as Sale -> Invoice -> Delivery).
For single-Model processes they are an overkill and can be replaced by a Kanban Board:

I would like to drop states and workflows, to replace them by Stages + Kanban Board.
History tracking and notifications are Stage will be there. Additional logic (mostly will be custom) is possible with Automated Actions.

bwrsandman replied:
@max3903 do you agree?

max3903 replied:
I am sorry I don't. Workflows are useful to trigger server actions (like sending notifications).

dreispt replied:
@max3903  Most common notifications are already triggered by mail_message tracking, and it's trivial to add additional messages or actions using Automated Actions. I have been using that extensively to: send reminder notifications, automatically route/assign issues to people, change kanban card colors to yellow or red or their priority, change block user actions on certain conditions.

I can share these implementation if you like. So I still don't see a use case where Workflows would be needed.

bealdav replied:
I don't know your case but workflow seems overkill general performance of the erp. No ?
@dreispt :+1

max3903 replied:
@dreispt Ok, can you provide the required data and configuration to:
- send a notification to the responsible user when the action is open
- send a reminder to the responsible user 10 days before the action deadline

dreispt replied:
@max3903 Agreed. The first one will be out of the box, through Stage tracking, the second will be a demo data example.

bwrsandman replied:
Is this blocking the current PR or can we add @dreispt's ideas in a following PR?

bwrsandman commented:
@dreispt Most other porting PRs are dependent on this one, do you think your adjustments can be made after merging this?

max3903 commented:
I think we added a name field on version 7 and I don't see it here.

Otherwise
:+1:

ghost commented:
Hi! @dreispt, @max3903 

May I include Spanish translation in this branch 8.0-mgmtsystem_nonconformity ?

max3903 commented:
@gs-clearcorp Sure!

Commit:
[FIX] - mgmtsystem_nonconformity: Fixed typo in data.

Commit:
[IMP] - mgmtsystem_nonconformity: Added updated es translation.

ghost commented:
PR here savoirfairelinux/management-system#18

Commit:
Merge pull request #18 from ClearCorp-dev/8.0-mgmtsystem_nonconformity-gs

8.0 mgmtsystem nonconformity gs

dreispt commented:
@bwrsandman @max3903 Good enough today is better than perfect tomorrow, so yes, go ahead.
@max3903 I won't be able do to work on this, so maybe the idea can be put into the roadmap.

max3903 commented:
@dreispt not sure what you want to add, but here it is #84 

:+1:

max3903 commented:
:+1:

dreispt commented:
:+1:
